### DISCUSSION TITLE
⌛ 🔗 Global reCAPTCHA: Secure All Your Forms with Ease

### DISCUSSION THREAD

**Discussion by nesoor** [elementor/elementor#24835] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

I have to manually set up reCAPTCHA for each form. However, I run a web design agency, and our clients also handle their own logins. Unfortunately, they often neglect to include reCAPTCHA, leading to complaints about form spam. I find myself repeatedly having to explain the importance of adding reCAPTCHA. Since they may not be familiar with all the technical aspects, I understand that they frequently forget. Nonetheless, as a web designer, it can be quite frustrating for me.

### Describe the solution you'd like.

It would be immensely beneficial to have the ability to activate a global reCAPTCHA setting, ensuring that every form we and the customer implement is automatically protected by Google reCAPTCHA. Such a feature would significantly reduce the workload and the need for extensive explanations on our part.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by toltechinternet** [comment]
I completely agree, with the fact we have to navigate to Elementor > Settings to input our reCAPTCHA secret and public keys, at this point we should have the option for this key to cover all our Elementor forms. I feel it's counterproductive to then navigate away from the settings and locate each form just to add Google reCAPTCHA multiple times (especially with websites that have more than one form).

If you look at Ninja Forms, they allow you to input the keys and your form is automatically covered with Google ReCAPTCHA sitewide.

**Comment by BastiaanBosch** [comment]
We have a problem with bots creating fake registrations.

What surprises me is that there is a reCAPTCHA for submission forms and login,
but not being able to create a secure registration form … to prevent bots from registering?

Preventing fake registrations, (username + password), is first line of defence,
is more important than preventing Robots / Hackers to guess username and password.
When robots are able to crate fake accounts they are already in…

So I created /user-registration-form/
with php script copy-paste from 
https://pluginsforwp.com/blog/elementor-registration-form/
and added hCaptcha https://docs.hcaptcha.com/ to the form.

Luckily I can copy-paste ;-)

But it does not work  “ :-( “  new user is not added.

Considering a secure registration form as primary WP / Elementor functionality, 
I contacted Elementor support whom redirected me to you.. The Elementor developer's forum
GitHub https://github.com/elementor/elementor.

The essential-addons.com plugin has login-register-form functionality as Widget,
However to add this plugin for that widget only is an CPU/RAM consuming “overkill” .


Required Functionality
1) user-registration-form  (with reCaptcha / hCaptcha)
After Submit
2) Added as Pending user (with full IP address, allowing Admins to manually block dangerous IP’s)
3) email is send with UUiD code to click for (2FA) authorization.
4) if clicked, user “role” changes from Pending to user (with selected role subscriber)

Feature Request: 
ADD create-new-user-registration-form Widget to the Elementor Roadmap?

Since the PHP code is already available and widely used, (but outside my scope of knowledge)
I think that the development can be very fast.

Since New AI Bots become smarter every day… 

I have considered to make this [ create-new-user-registration-form Widget ] feature request a new request,
But since it is related to the reCAPTCHA, i decided to add it here as response to add.
When I am wrong in this, please moderate it into a new request.
